client/src/main/java/org/apache/uniffle/client/record/RecordBuffer.java

```diff
@@ -69,7 +69,9 @@ public void sort(Comparator comparator) {
         new Comparator<Record>() {
           @Override
           public int compare(Record o1, Record o2) {
-            return comparator.compare(o1.getKey(), o2.getKey());
+            int h1 = (o1 == null) ? 0 : o1.hashCode();
```

Comment:
It seems that the change has disrupted the original semantics of "sort by record key".

Reply:
Hi @LuciferYang 
Thanks for the reminder and I updated the part you mentioned. 😄